src/sources/soundsourcem4a.cpp

```diff
@@ -372,213 +510,308 @@ ReadableSampleFrames SoundSourceM4A::readSampleFramesClamped(
-            if (isValidSampleBlockId(m_curSampleBlockId)) {
-                // Read data for next sample block into input buffer
-                u_int8_t* pInputBuffer = &m_inputBuffer[0];
-                u_int32_t inputBufferLength = m_inputBuffer.size(); // in/out parameter
-                if (!MP4ReadSample(m_hFile, m_trackId, m_curSampleBlockId, &pInputBuffer, &inputBufferLength, nullptr, nullptr, nullptr, nullptr)) {
+            DEBUG_ASSERT(m_inputBufferLength >= m_inputBufferOffset);
+            if (m_inputBufferLength <= m_inputBufferOffset) {
+                break; // EOF
+            }
+
+            // NOTE(uklotzde): The sample buffer for Decode2 has to
+            // be big enough for a whole block of decoded samples, which
+            // contains up to m_framesPerSampleBlock frames. Otherwise
+            // we need to use a temporary buffer.
+            CSAMPLE* pDecodeBuffer; // in/out parameter
+            SINT decodeBufferCapacity;
+            const SINT decodeBufferCapacityMin =
+                    getSignalInfo().frames2samples(m_framesPerSampleBlock);
+            if (writableSampleFrames.writableData() &&
+                    (decodeBufferCapacityMin <= numberOfSamplesRemaining)) {
+                // Decode samples directly into the output buffer
+                pDecodeBuffer =
+                        writableSampleFrames.writableData(outputSampleOffset);
+                decodeBufferCapacity = numberOfSamplesRemaining;
+            } else {
+                // Decode next sample block into temporary buffer
+                const SINT maxWriteLength = math_max(
+                        numberOfSamplesRemaining, decodeBufferCapacityMin);
+                const SampleBuffer::WritableSlice writableSlice(
+                        m_sampleBuffer.growForWriting(maxWriteLength));
+                pDecodeBuffer = writableSlice.data();
+                decodeBufferCapacity = writableSlice.length();
+            }
+            DEBUG_ASSERT(decodeBufferCapacityMin <= decodeBufferCapacity);
+
+            faad2::FrameInfo decFrameInfo;
+            DEBUG_ASSERT(m_inputBufferLength >= m_inputBufferOffset);
+            void* pDecodeResult = m_pFaad->Decode2(m_hDecoder,
+                    &decFrameInfo,
+                    &m_inputBuffer[m_inputBufferOffset],
+                    m_inputBufferLength - m_inputBufferOffset,
+                    reinterpret_cast<void**>(&pDecodeBuffer),
+                    decodeBufferCapacity * sizeof(*pDecodeBuffer));
+            if (decFrameInfo.error != 0) {
+                // A decoding error has occurred
+                if (retryAfterReopeningDecoder) {
+                    // At this point we have failed to decode the current sample
+                    // block twice and need to discard it. The content of the
+                    // sample block is unknown and we simply continue with the
+                    // next block. This is just a workaround! The reason why FAAD2
```

Comment:
by using silence instead  ..
or 
by skipping the track. 

I am afraid that the later will produce a cue point offset if the track was originally decoded with a working FAAD2 decoder or if it was decoded with a broken decoder the offset will appear after the decoder is fixed.

Reply:
I am not writing an AAC decoder for the unknown frames. No chance. In this case I will close this PR.

Reply:
We don't even know if and how many samples those blocks contain. Sorry, but this is not doable.